src/components/NameFilter/NameFilter.tsx

```diff
@@ -0,0 +1,31 @@
+import { useSearchParams } from 'react-router-dom';
+
+export const NameFilter = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+
+  return (
+    <p className="control has-icons-left">
+      <input
+        className="input"
+        type="text"
+        placeholder="Search"
+        value={query}
+        onChange={event => {
+          const newQuery = event.target.value;
+
```

Comment:
I think it's better to make a callback function for this purpose, because you have conditions here

Reply:
totally agree, let's make a callback here and in similar cases